internal/controller/validatorconfig_controller.go

```diff
@@ -284,8 +286,15 @@ func (r *ValidatorConfigReconciler) deletePlugins(ctx context.Context, vc *v1alp
 		if release.Secret.Labels == nil || release.Secret.Labels["owner"] != "helm" {
 			return nil
 		}
-		r.deletePlugin(vc, p.Chart.Name)
+
+		wg.Add(1)
+		go func(name string) {
+			defer wg.Done()
+			r.deletePlugin(vc, name)
+		}(p.Chart.Name)
```

Comment:
i was initially looking to put everything inside this for loop (L278) in a goroutine, but doing that resulted in a lot of unexpected errors. The safest way to resolve the timeout issues was to just parallelize the actual plugin deletion call.

given we add enough plugins, we're probably still at risk for hitting the same timeout related errors we initially saw, so it might still make sense to bump the timeout in the spectro-cleanup job. not sure yet how much headroom we have before this becomes an issue again though.

Reply:
yep this change matches what I had in mind.. looks good!